src/map/entities/battleentity.cpp

```diff
@@ -388,12 +388,7 @@ uint8 CBattleEntity::UpdateSpeed(bool run)
     // TODO: Find exceptions. Add them here.
 
     // GM speed bypass.
-
-    if (getMod(Mod::MOVE_SPEED_OVERRIDE) > 255)
-    {
-        outputSpeed = 0;
-    }
-    else if (getMod(Mod::MOVE_SPEED_OVERRIDE) > 0)
```

Comment:
I would really like to keep this functionality

Reply:
setBaseSpeed(0) will do this now if I understand correctly

Reply:
> setBaseSpeed(0) will do this now if I understand correctly

This is the intent yes but there is one issue I just noticed. If base speed is set to 0, bolter's roll can still raise it. Instead, I can just return 0 if base speed is 0 (this wouldn't affect setting your speed to 0 with gear or other means). I would prefer something like that, where "base speed 0 means 0 full stop" as opposed to "big number means stop". IDK if "magic functionality" is a real term but that's what this feels like to me whereas setting base speed to 0 makes more logical sense.

Reply:
Something like this:
```cpp
    if (getMod(Mod::MOVE_SPEED_OVERRIDE) > 0)
    {
        outputSpeed = getMod(Mod::MOVE_SPEED_OVERRIDE);
    }
    else if (baseSpeed == 0)
    {
        return baseSpeed;
    }
    ```

Reply:
How about this:
```c++
    int16 = speedOverride = getMod(Mod::MOVE_SPEED_OVERRIDE)

    if (speedOverride > 255 || baseSpeed == 0)
    {
        return 0;
    }
    else if (speedOverride > 0)
    {
        return speedOverride;
    }
```

Reply:
Edited

Reply:
What's the reason for keeping the mod like that though when setting base speed would do the same thing?

Reply:
- It doesnt hurt, considering that if that modifier went over 255 it would overflow the speed output anyway
- It allows for GMs to set their speed to 0 using the !speed GM command instead of anything else
- Its a single line of code

Reply:
I mean, it's clamped so it would never overflow. The GM command is a valid point I guess, I just don't like that pattern is all. Would it also be acceptable to have a negative modifier set it to 255? It's one of those things where if you know about it, you can use it instead of just setting the base speed, but if you don't know about it, you unexpectedly get bound if you go over.

Reply:
Rearranged some stuff to reduce work being done, relevant part here:
```cpp
    else if (baseSpeed == 0 || getMod(Mod::MOVE_SPEED_OVERRIDE) > 255)
    {
        outputSpeed = 0;
    }
    else if (getMod(Mod::MOVE_SPEED_OVERRIDE) > 0)
    {
        // GM speed bypass.
        // Speed cap can be bypassed. Ex. Feast of swords. GM speed.
        // TODO: Find exceptions. Add them here.
        outputSpeed = getMod(Mod::MOVE_SPEED_OVERRIDE);
    }
    else
    {
```
everything else is just indentation.

Reply:
@Xaver-DaRed how do you feel about this second commit? Using a negative mod to bind at 0 speed, otherwise clamping 0-255. Functionally it's the same, but is more intuitive imo, and fixes the remaining issues with the speed command.

Reply:
to me -1 does indicate something special,  you wouldn't normally set a speed of -1 after all

I have no objections